docs/aggregation.md

```diff
@@ -0,0 +1,49 @@
+# Aggregations
+
+We will cover internal implementation of common aggregations in this document.
+
+
+## Frontend
+
+TODO
+
+## Expression Framework
+
+TODO
+
+## HashAggExecutor
+
+![aggregation components](./images/aggregation/agg-components.png)
+
+Within the `HashAggExecutor`, there are 4 main components:
+1. AggCalls.
+2. AggState.
+3. AggGroups.
+4. Persisted State.
+
+AggCalls are the aggregation calls for the query. For instance `SUM(v1)`, `COUNT(v2)` has the AggCalls `SUM` and `COUNT`.
+
+AggState is the state we use to compute to the result (output) of the aggregation call.
+Within each aggregation group, it will have an AggState for each AggCall.
+
+AggGroups are created per aggregation group.
+For instance with `GROUP BY v1, v2`, there will be a group for each unique combination of `v1` and `v2`.
```

Comment:
May be better to change the column names cuz they are used before as agg call arguments. Can cause confusion.